### ISSUE TITLE
Multiple Issues

### ISSUE THREAD

**Issue by SonnenPingu** [Zrips/CMI#8757]
### Description of Issue

1:
! [Mon 10:35:47 WARN ] java.lang.IllegalStateException: BlockReceiveGameEvent may only be triggered synchronously.
! [Mon 10:35:47 WARN ]     at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:42)
! [Mon 10:35:47 WARN ]     at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:126)
! [Mon 10:35:47 WARN ]     at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:615)
! [Mon 10:35:47 WARN ]     at net.minecraft.world.level.gameevent.vibrations.VibrationSystem$b.a(VibrationSystem.java:296)
! [Mon 10:35:47 WARN ]     at net.minecraft.world.level.gameevent.GameEventDispatcher.lambda$post$0(GameEventDispatcher.java:50)
! [Mon 10:35:47 WARN ]     at net.minecraft.world.level.gameevent.EuclideanGameEventListenerRegistry.a(EuclideanGameEventListenerRegistry.java:73)
! [Mon 10:35:47 WARN ]     at net.minecraft.world.level.gameevent.GameEventDispatcher.a(GameEventDispatcher.java:62)
! [Mon 10:35:47 WARN ]     at net.minecraft.server.level.WorldServer.a(WorldServer.java:1844)
! [Mon 10:35:47 WARN ]     at net.minecraft.world.level.GeneratorAccess.a(GeneratorAccess.java:107)
! [Mon 10:35:47 WARN ]     at net.minecraft.world.entity.Entity.a(Entity.java:1673)
! [Mon 10:35:47 WARN ]     at net.minecraft.world.entity.Entity.p(Entity.java:3021)
! [Mon 10:35:47 WARN ]     at net.minecraft.world.entity.Entity.a(Entity.java:2962)
! [Mon 10:35:47 WARN ]     at org.bukkit.craftbukkit.v1_20_R3.entity.CraftEntity.addPassenger(CraftEntity.java:441)
! [Mon 10:35:47 WARN ]     at CMI-9.7.2.10.jar//com.Zrips.CMI.Modules.Chat.CMIChatBubble.<init>(CMIChatBubble.java:59)
! [Mon 10:35:47 WARN ]     at CMI-9.7.2.10.jar//com.Zrips.CMI.Modules.Chat.ChatBubbleManager.showBubble(ChatBubbleManager.java:110)
 
! [Mon 10:35:47 WARN ]     at CMI-9.7.2.10.jar//com.Zrips.CMI.Modules.NickName.NickNameListener.AsyncPlayerChatEventRawConverter(NickNameListener.java:209)
! [Mon 10:35:47 WARN ]     at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor826.execute(Unknown Source)
! [Mon 10:35:47 WARN ]     at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:77)
! [Mon 10:35:47 WARN ]     at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:77)
! [Mon 10:35:47 WARN ]     at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70)
! [Mon 10:35:47 WARN ]     at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54)
! [Mon 10:35:47 WARN ]     at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:126)
! [Mon 10:35:47 WARN ]     at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:615)
! [Mon 10:35:47 WARN ]     at io.papermc.paper.adventure.ChatProcessor.post(ChatProcessor.java:378)
! [Mon 10:35:47 WARN ]     at io.papermc.paper.adventure.ChatProcessor.process(ChatProcessor.java:85)
! [Mon 10:35:47 WARN ]     at net.minecraft.server.network.PlayerConnection.chat(PlayerConnection.java:2318)
! [Mon 10:35:47 WARN ]     at net.minecraft.server.network.PlayerConnection.b(PlayerConnection.java:2447)
! [Mon 10:35:47 WARN ]     at net.minecraft.server.network.PlayerConnection.lambda$handleChat$13(PlayerConnection.java:2167)
! [Mon 10:35:47 WARN ]     at net.minecraft.util.FutureChain.a(FutureChain.java:25)
! [Mon 10:35:47 WARN ]     at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(Unknown Source)
! [Mon 10:35:47 WARN ]     at java.base/java.util.concurrent.CompletableFuture$Completion.run(Unknown Source)
! [Mon 10:35:47 WARN ]     at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
! [Mon 10:35:47 WARN ]     at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
! [Mon 10:35:47 WARN ]     at java.base/java.lang.Thread.run(Unknown Source)
2:

[11:41:21] [User Authenticator #3/INFO]: UUID of player Gamermariene is 169753c7-2ceb-470f-9d70-78be3a1e5353
[11:41:22] [Server thread/WARN]: java.lang.NullPointerException: Cannot invoke "org.bukkit.World.getUID()" because the return value of "org.bukkit.Location.getWorld()" is null
[11:41:22] [Server thread/WARN]: 	at CMI-9.7.2.10.jar//com.Zrips.CMI.Modules.Teleportations.TeleportHandler.teleportPlayer(TeleportHandler.java:361)
[11:41:22] [Server thread/WARN]: 	at CMI-9.7.2.10.jar//com.Zrips.CMI.Modules.Teleportations.TeleportHandler.teleportPlayer(TeleportHandler.java:302)
[11:41:22] [Server thread/WARN]: 	at CMI-9.7.2.10.jar//com.Zrips.CMI.AllListeners.PlayerListeners.onPlayerFirstJoin(PlayerListeners.java:1088)
[11:41:22] [Server thread/WARN]: 	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor647.execute(Unknown Source)
[11:41:22] [Server thread/WARN]: 	at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:77)
[11:41:22] [Server thread/WARN]: 	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:77)
[11:41:22] [Server thread/WARN]: 	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70)
[11:41:22] [Server thread/WARN]: 	at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54)
[11:41:22] [Server thread/WARN]: 	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:126)
[11:41:22] [Server thread/WARN]: 	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:615)
[11:41:22] [Server thread/WARN]: 	at net.minecraft.server.players.PlayerList.a(PlayerList.java:345)
[11:41:22] [Server thread/WARN]: 	at net.minecraft.server.network.ServerConfigurationPacketListenerImpl.a(ServerConfigurationPacketListenerImpl.java:134)
[11:41:22] [Server thread/WARN]: 	at net.minecraft.network.protocol.configuration.ServerboundFinishConfigurationPacket.a(ServerboundFinishConfigurationPacket.java:18)
[11:41:22] [Server thread/WARN]: 	at net.minecraft.network.protocol.configuration.ServerboundFinishConfigurationPacket.a(ServerboundFinishConfigurationPacket.java:9)
[11:41:22] [Server thread/WARN]: 	at net.minecraft.network.protocol.PlayerConnectionUtils.lambda$ensureRunningOnSameThread$0(PlayerConnectionUtils.java:54)
[11:41:22] [Server thread/WARN]: 	at net.minecraft.server.TickTask.run(TickTask.java:18)
[11:41:22] [Server thread/WARN]: 	at net.minecraft.util.thread.IAsyncTaskHandler.d(IAsyncTaskHandler.java:149)
[11:41:22] [Server thread/WARN]: 	at net.minecraft.util.thread.IAsyncTaskHandlerReentrant.d(IAsyncTaskHandlerReentrant.java:24)
[11:41:22] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1465)
[11:41:22] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:194)
[11:41:22] [Server thread/WARN]: 	at net.minecraft.util.thread.IAsyncTaskHandler.x(IAsyncTaskHandler.java:123)
[11:41:22] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.bl(MinecraftServer.java:1442)
[11:41:22] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1365)
[11:41:22] [Server thread/WARN]: 	at net.minecraft.util.thread.IAsyncTaskHandler.c(IAsyncTaskHandler.java:133)
[11:41:22] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.w_(MinecraftServer.java:1343)
[11:41:22] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1232)
[11:41:22] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:319)
[11:41:22] [Server thread/WARN]: 	at java.base/java.lang.Thread.run(Unknown Source)

3: 

! [Sun 23:34:27 WARN ] java.lang.NullPointerException: Cannot invoke "org.bukkit.World.getUID()" because the return value of "org.bukkit.Location.getWorld()" is null
 
! [Sun 23:34:27 WARN ]     at CMI-9.7.2.10.jar//com.Zrips.CMI.Modules.Teleportations.TeleportHandler.teleportPlayer(TeleportHandler.java:361)
! [Sun 23:34:27 WARN ]     at CMI-9.7.2.10.jar//com.Zrips.CMI.Modules.Teleportations.TeleportHandler.teleportPlayer(TeleportHandler.java:314)
! [Sun 23:34:27 WARN ]     at CMI-9.7.2.10.jar//com.Zrips.CMI.commands.list.spawn.perform(spawn.java:58)
! [Sun 23:34:27 WARN ]     at CMI-9.7.2.10.jar//com.Zrips.CMI.commands.CommandsHandler.onCommand(CommandsHandler.java:400)
! [Sun 23:34:27 WARN ]     at CMI-9.7.2.10.jar//com.Zrips.CMI.Containers.CommandReg$ReflectCommand.execute(CommandReg.java:231)
! [Sun 23:34:27 WARN ]     at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:155)
! [Sun 23:34:27 WARN ]     at org.bukkit.craftbukkit.v1_20_R3.CraftServer.dispatchCommand(CraftServer.java:999)
! [Sun 23:34:27 WARN ]     at org.bukkit.craftbukkit.v1_20_R3.command.BukkitCommandWrapper.run(BukkitCommandWrapper.java:64)
! [Sun 23:34:27 WARN ]     at com.mojang.brigadier.context.ContextChain.runExecutable(ContextChain.java:73)
! [Sun 23:34:27 WARN ]     at net.minecraft.commands.execution.tasks.ExecuteCommand.a(ExecuteCommand.java:31)
! [Sun 23:34:27 WARN ]     at net.minecraft.commands.execution.tasks.ExecuteCommand.execute(ExecuteCommand.java:19)
! [Sun 23:34:27 WARN ]     at net.minecraft.commands.execution.UnboundEntryAction.a(UnboundEntryAction.java:8)
! [Sun 23:34:27 WARN ]     at net.minecraft.commands.execution.CommandQueueEntry.a(CommandQueueEntry.java:5)
! [Sun 23:34:27 WARN ]     at net.minecraft.commands.execution.ExecutionContext.a(ExecutionContext.java:103)
! [Sun 23:34:27 WARN ]     at net.minecraft.commands.CommandDispatcher.a(CommandDispatcher.java:434)
! [Sun 23:34:27 WARN ]     at net.minecraft.commands.CommandDispatcher.performCommand(CommandDispatcher.java:336)
! [Sun 23:34:27 WARN ]     at net.minecraft.commands.CommandDispatcher.a(CommandDispatcher.java:323)
! [Sun 23:34:27 WARN ]     at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:2230)
! [Sun 23:34:27 WARN ]     at net.minecraft.server.network.PlayerConnection.lambda$handleChatCommand$14(PlayerConnection.java:2190)
! [Sun 23:34:27 WARN ]     at net.minecraft.util.thread.IAsyncTaskHandler.b(IAsyncTaskHandler.java:59)
! [Sun 23:34:27 WARN ]     at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source)
! [Sun 23:34:27 WARN ]     at net.minecraft.server.TickTask.run(TickTask.java:18)
! [Sun 23:34:27 WARN ]     at net.minecraft.util.thread.IAsyncTaskHandler.d(IAsyncTaskHandler.java:149)
! [Sun 23:34:27 WARN ]     at net.minecraft.util.thread.IAsyncTaskHandlerReentrant.d(IAsyncTaskHandlerReentrant.java:24)
! [Sun 23:34:27 WARN ]     at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1465)
! [Sun 23:34:27 WARN ]     at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:194)
! [Sun 23:34:27 WARN ]     at net.minecraft.util.thread.IAsyncTaskHandler.x(IAsyncTaskHandler.java:123)
! [Sun 23:34:27 WARN ]     at net.minecraft.server.MinecraftServer.bl(MinecraftServer.java:1442)
! [Sun 23:34:27 WARN ]     at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1365)
! [Sun 23:34:27 WARN ]     at net.minecraft.util.thread.IAsyncTaskHandler.bu(IAsyncTaskHandler.java:112)
! [Sun 23:34:27 WARN ]     at net.minecraft.server.MinecraftServer.a(MinecraftServer.java:1557)
! [Sun 23:34:27 WARN ]     at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1226)
! [Sun 23:34:27 WARN ]     at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:319)
! [Sun 23:34:27 WARN ]     at java.base/java.lang.Thread.run(Unknown Source)
! [Sun 23:34:27 WARN ] java.lang.NullPointerException: Cannot invoke "org.bukkit.craftbukkit.v1_20_R3.CraftWorld.getHandle()" because the return value of "org.bukkit.Location.getWorld()" is null
! [Sun 23:34:27 WARN ]     at org.bukkit.craftbukkit.v1_20_R3.entity.CraftEntity.teleportAsync(CraftEntity.java:1035)
! [Sun 23:34:27 WARN ]     at org.bukkit.entity.Entity.teleportAsync(Entity.java:213)
! [Sun 23:34:27 WARN ]     at CMILib1.4.7.10.jar//net.Zrips.CMILib.Version.PaperMethods.AsyncTeleportPaper.teleportAsync(AsyncTeleportPaper.java:16)
! [Sun 23:34:27 WARN ]     at CMILib1.4.7.10.jar//net.Zrips.CMILib.Version.PaperMethods.Environment.teleport(Environment.java:38)
! [Sun 23:34:27 WARN ]     at CMILib1.4.7.10.jar//net.Zrips.CMILib.Version.PaperMethods.PaperLib.teleportAsync(PaperLib.java:98)
! [Sun 23:34:27 WARN ]     at CMI-9.7.2.10.jar//com.Zrips.CMI.Modules.Teleportations.TeleportHandler.finalizeTeleportation(TeleportHandler.java:346)
! [Sun 23:34:27 WARN ]     at CMI-9.7.2.10.jar//com.Zrips.CMI.Modules.Teleportations.TeleportHandler.lambda$10(TeleportHandler.java:424)
! [Sun 23:34:27 WARN ]     at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(Unknown Source)
! [Sun 23:34:27 WARN ]     at java.base/java.util.concurrent.CompletableFuture$Completion.exec(Unknown Source)
! [Sun 23:34:27 WARN ]     at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
! [Sun 23:34:27 WARN ]     at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
! [Sun 23:34:27 WARN ]     at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
! [Sun 23:34:27 WARN ]     at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
! [Sun 23:34:27 WARN ]     at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source) 

4: (Jobs issue)

### Version Information

```shell
[Mon 12:05:09 INFO ] --------------------------------------------------
  [Mon 12:05:09 INFO ] CMI Plugin Version: 9.7.2.10 Velocity CMIB  SqLite
  [Mon 12:05:09 INFO ] CMILib: 1.4.7.11 
  [Mon 12:05:09 INFO ] Server: Paper(496) 1.20.4-R0.1-SNAPSHOT-
  [Mon 12:05:09 INFO ] CMI economy: Aktiviert Vault: 1.7.3-CMI CMI Chat: Aktiviert 
  [Mon 12:05:09 INFO ] Modules -> 53 enabled 6 disabled: tablist, ranks, votifier, shulkerBackpack, holograms, jail
  [Mon 12:05:09 INFO ] --------------------------------------------------
```

### Errors

```shell
1: https://mclo.gs/uzWJ9Aa#L321
2: https://mclo.gs/6GUnk7k#L1133
3:https://mclo.gs/icMvcvK#L3159
4: https://mclo.gs/icMvcvK#L465 its a Jobs issue
```

### Relevant Plugins

CMI Plugin Version: 9.7.2.10 Velocity CMIB  SqLite CMILib: 1.4.7.11 

### Agreements

- [X] My server is supported by CMI.
- [X] My version of CMI at the time of this report is up to date.
- [X] I have searched the github and asked around before making this report.

**Comment by SonnenPingu** [comment]
```diff
! [Tue 22:02:17 WARN ] 	at CMI-9.7.2.10.jar//com.Zrips.CMI.Modules.NickName.NickNameListener.AsyncPlayerChatEventRawConverter(NickNameListener.java:209)
! [Tue 22:02:17 WARN ] 	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor826.execute(Unknown Source)
! [Tue 22:02:17 WARN ] 	at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:77)
! [Tue 22:02:17 WARN ] 	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:77)
! [Tue 22:02:17 WARN ] 	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70)
! [Tue 22:02:17 WARN ] 	at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54)
! [Tue 22:02:17 WARN ] 	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:126)
! [Tue 22:02:17 WARN ] 	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:615)
! [Tue 22:02:17 WARN ] 	at io.papermc.paper.adventure.ChatProcessor.post(ChatProcessor.java:378)
! [Tue 22:02:17 WARN ] 	at io.papermc.paper.adventure.ChatProcessor.process(ChatProcessor.java:85)
! [Tue 22:02:17 WARN ] 	at net.minecraft.server.network.PlayerConnection.chat(PlayerConnection.java:2318)
! [Tue 22:02:17 WARN ] 	at net.minecraft.server.network.PlayerConnection.b(PlayerConnection.java:2447)
! [Tue 22:02:17 WARN ] 	at net.minecraft.server.network.PlayerConnection.lambda$handleChat$13(PlayerConnection.java:2167)
! [Tue 22:02:17 WARN ] 	at net.minecraft.util.FutureChain.a(FutureChain.java:25)
! [Tue 22:02:17 WARN ] 	at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(Unknown Source)
! [Tue 22:02:17 WARN ] 	at java.base/java.util.concurrent.CompletableFuture$Completion.run(Unknown Source)
! [Tue 22:02:17 WARN ] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
! [Tue 22:02:17 WARN ] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)```

**Comment by Zrips** [comment]
Both issues relating to same chatbubble error which should have been fixed with 9.7.2.11 version
